contracts/examples/ExampleCombinedSwapAddRemoveLiquidity.sol

```diff
@@ -26,6 +27,13 @@ contract ExampleCombinedSwapAddRemoveLiquidity {
         weth = weth_;
     }
 
+    // grants unlimited approval for a token to the router unless the existing allowance is high enough
+    function approveRouter(address _token, uint256 _amount) internal {
+        if (IERC20(_token).allowance(address(this), address(router)) < _amount) {
+            TransferHelper.safeApprove(_token, address(router), uint256(-1));
```

Comment:
don't we have to approve(0) first?

Reply:
Yes, indeed. I had assumed that tokens that forbid calling `approve()` with a non-zero amount when the existing allowance is non-zero would skip the allowance update in `transferFrom()` when the allowance is MAX_UINT (as is the case for USDT) but thinking about this more, I see no reason to make this assumption. Will fix.